File: configuration/src/main/java/com/synopsys/integration/configuration/property/types/integer/IntegerValueParser.java
package com.synopsys.integration.configuration.property.types.integer;

import org.jetbrains.annotations.NotNull;

import com.synopsys.integration.configuration.parse.ValueParseException;
import com.synopsys.integration.configuration.parse.ValueParser;

class IntegerValueParser extends ValueParser<Integer> {
    @NotNull
    @Override
    public Integer parse(@NotNull String value) throws ValueParseException {
        try {
            return Integer.parseInt(value);
        } catch (NumberFormatException e) {
            throw new ValueParseException(value, "integer", e);
        }
    }
}

File: configuration/src/main/java/com/synopsys/integration/configuration/property/types/integer/IntegerValueParser.kt
package com.synopsys.integration.configuration.property.types.integer

import com.synopsys.integration.configuration.parse.ValueParseException
import com.synopsys.integration.configuration.parse.ValueParser

class IntegerValueParser : ValueParser<Int>() {
    override fun parse(value: String) : Int {
        return try {
            value.toInt()
        } catch (e: NumberFormatException) {
            throw ValueParseException(value, "integer", innerException = e)
        }
    }
}